packages/e2e-tests/specs/block-grouping.test.js

```diff
@@ -0,0 +1,234 @@
+	beforeAll( async () => {
+		await createNewPost();
+	} );
+
+	beforeEach( async () => {
+		// Remove all blocks from the post so that we're working with a clean slate
+		await page.evaluate( () => {
+			const blocks = wp.data.select( 'core/editor' ).getBlocks();
+			const clientIds = blocks.map( ( block ) => block.clientId );
+			wp.data.dispatch( 'core/editor' ).removeBlocks( clientIds );
+		} );
+	} );
+
+	describe( 'Group creation', () => {
+		it( 'creates a group from multiple blocks of the same type via block transforms', async () => {
+			// Creating test blocks
+			await insertBlocksOfSameType();
+
+			// Multiselect via keyboard.
+			await pressKeyWithModifier( 'primary', 'a' );
+			await pressKeyWithModifier( 'primary', 'a' );
+
+			await transformBlockTo( 'Group' );
+
+			expect( await getEditedPostContent() ).toMatchSnapshot();
+		} );
+
+		it( 'creates a group from multiple blocks of different types via block transforms', async () => {
+			// Creating test blocks
+			await insertBlocksOfMultipleTypes();
+
+			// Multiselect via keyboard.
+			await pressKeyWithModifier( 'primary', 'a' );
+			await pressKeyWithModifier( 'primary', 'a' );
+
+			await transformBlockTo( 'Group' );
+
+			expect( await getEditedPostContent() ).toMatchSnapshot();
+		} );
+
+		it( 'creates a group from multiple blocks of the same type via options toolbar', async () => {
+			// Creating test blocks
+			await insertBlocksOfSameType();
+
+			// Multiselect via keyboard.
+			await pressKeyWithModifier( 'primary', 'a' );
+			await pressKeyWithModifier( 'primary', 'a' );
+
+			await clickBlockToolbarButton( 'More options' );
+
+			const groupButton = await page.waitForXPath( '//button[text()="Group"]' );
+			await groupButton.click();
+
+			expect( await getEditedPostContent() ).toMatchSnapshot();
+		} );
+
+		it( 'creates a group from multiple blocks of different types via options toolbar', async () => {
+			// Creating test blocks
+			await insertBlocksOfMultipleTypes();
+
+			// Multiselect via keyboard.
+			await pressKeyWithModifier( 'primary', 'a' );
+			await pressKeyWithModifier( 'primary', 'a' );
+
+			await clickBlockToolbarButton( 'More options' );
+
+			const groupButton = await page.waitForXPath( '//button[text()="Group"]' );
+			await groupButton.click();
+
+			expect( await getEditedPostContent() ).toMatchSnapshot();
+		} );
+	} );
+
+	describe( 'Ungrouping', () => {
+		it( 'ungroups an existing group via options menu', async () => {
+			// Create a Group
+			await insertBlocksOfMultipleTypes();
+			await pressKeyWithModifier( 'primary', 'a' );
+			await pressKeyWithModifier( 'primary', 'a' );
+			await transformBlockTo( 'Group' );
+
+			await clickBlockToolbarButton( 'More options' );
+
+			const unGroupButton = await page.waitForXPath( '//button[text()="Ungroup"]' );
+
+			await unGroupButton.click();
+
+			expect( await getEditedPostContent() ).toMatchSnapshot();
+		} );
+	} );
+
+	describe( 'Keyboard shortcuts', () => {
```

Comment:
The two test cases in this section could also be combined into one.

Reply:
Fixed